pythran/pythonic/numpy/partial_sum.hpp

```diff
@@ -13,24 +12,53 @@ namespace pythonic
   namespace numpy
   {
 
-    template <class Op>
-    template <class E, class F, class A>
-    A _partial_sum<Op, 1>::operator()(E e, F &o, A acc)
+    template <class Op, class A>
+    template <class E, class F>
+    A _partial_sum<Op, 1, A>::operator()(E e, F &o, A acc)
     {
       for (auto const &value : e) {
-        Op{}(acc, value);
+        acc = Op{}(acc, (A)value);
```

Comment:
On Mon, Aug 22, 2016 at 01:23:54PM -0700, pbrunet wrote:
> >      {
> >        for (auto const &value : e) {
> > -        Op{}(acc, value);
> > +        acc = Op{}(acc, (A)value);
> 
> Still an issue with boolean operations. With "equal" again, ccc may be [True, True] and value [1, 2], then equal(acc, value) return [True, True] while it will return [False, False] withotut the cast. I first though it was numpy.equal which behave this way but indead it is only for accumulation in numpy.

What's the signature of Op{} in that case ? If it's ool (bool, bool)
then there's no issue, right ?


Looks like a bad signature of Op...